addons/arrays/test_find.sqf

```diff
@@ -0,0 +1,185 @@
+// ----------------------------------------------------------------------------
+#define DEBUG_MODE_FULL
+#include "script_component.hpp"
+
+SCRIPT(test_find);
+
+// ----------------------------------------------------------------------------
+
+private ["_original", "_expected", "_result", "_fn"];
+
+_fn = "CBA_fnc_findNil";
+LOG("Testing " + _fn);
+
+TEST_DEFINED("CBA_fnc_findNil","");
+
+// Find First Nil Entry
+_original = ["", player, "", player, nil, "", player, nil];
+_result = ["", player, "", player, nil, "", player, nil] call CBA_fnc_findNil;
+_expected = 4;
+TEST_OP(str _original,==,str _expected,_fn);
+TEST_OP(str _result,==,str _expected,_fn);
+
+
+_original = ["", nil, "", player, nil, "", player, nil];
+_result = ["", nil, "", player, nil, "", player, nil] call CBA_fnc_findNil;
+_expected = 1;
+TEST_OP(str _original,==,str _expected,_fn);
+TEST_OP(str _result,==,str _expected,_fn);
+
+
+_original = ["", player, "", nil, nil, "", player, nil];
+_result = ["", player, "", nil, nil, "", player, nil] call CBA_fnc_findNil;
+_expected = 3;
+TEST_OP(str _original,==,str _expected,_fn);
+TEST_OP(str _result,==,str _expected,_fn);
+
+
+_original = ["", player, "", player, "", "", player, nil];
+_result = ["", player, "", player, "", "", player, nil] call CBA_fnc_findNil;
+_expected = 7;
+TEST_OP(str _original,==,str _expected,_fn);
+TEST_OP(str _result,==,str _expected,_fn);
+
+
+_original = ["", player, "", player, "", "", player, ""];
+_result = ["", player, "", player, "", "", player, ""] call CBA_fnc_findNil;
+_expected = -1;
+TEST_OP(str _original,==,str _expected,_fn);
+TEST_OP(str _result,==,str _expected,_fn);
+
+
+
+
+_fn = "CBA_fnc_findNull";
+LOG("Testing " + _fn);
+
+TEST_DEFINED("CBA_fnc_findNull","");
+
+
+// Find First Null Entry
+_original = ["", player, "", player, objNull, "", player, displayNull];
+_result = ["", player, "", player, objNull, "", player, displayNull] call CBA_fnc_findNull;
+_expected = 4;
+TEST_OP(str _original,==,str _expected,_fn);
+TEST_OP(str _result,==,str _expected,_fn);
+
+
+_original = ["", grpNull, "", player, objNull, "", player, displayNull];
+_result = ["", grpNull, "", player, objNull, "", player, displayNull] call CBA_fnc_findNull;
+_expected = 1;
+TEST_OP(str _original,==,str _expected,_fn);
+TEST_OP(str _result,==,str _expected,_fn);
+
+
+_original = ["", player, "",  scriptNull, objNull, "", player, displayNull];
+_result = ["", player, "",  scriptNull, objNull, "", player, displayNull] call CBA_fnc_findNull;
+_expected = 3;
+TEST_OP(str _original,==,str _expected,_fn);
+TEST_OP(str _result,==,str _expected,_fn);
+
+
+_original = ["", player, "", player, "", "", player, displayNull];
+_result = ["", player, "", player, "", "", player, displayNull] call CBA_fnc_findNull;
+_expected = 7;
+TEST_OP(str _original,==,str _expected,_fn);
+TEST_OP(str _result,==,str _expected,_fn);
+
```

Comment:
Good - these four tests above test for different kinds of null values (but a test with controlNull is missing). It also needs tests for
* an input array with a variable in it somewhere that is null-valued (for example objNull).
* an input array with a code element in it that returns one of the null values.
* an input array that is empty.